streams/src/main/java/org/apache/kafka/streams/internals/ApiUtils.java

```diff
@@ -75,4 +81,31 @@ public static long validateMillisecondInstant(final Instant instant, final Strin
     public static String prepareMillisCheckFailMsgPrefix(final Object value, final String name) {
         return format(MILLISECOND_VALIDATION_FAIL_MSG_FRMT, name, value);
     }
+
+    /**
+     * @throws IllegalArgumentException if the same instance is obtained each time
+     */
+    public static void checkSupplier(final Supplier<?> supplier) {
+        if (supplier.get() == supplier.get()) {
+            final String supplierClass = getAllImplementedInterfaces(supplier.getClass()).stream()
```

Comment:
Do we really need to try to extract the concrete interface name?

Reply:
We don't _really_ need to. I thought it could make for a better, and also predictable, error message. But maybe just using the implementing class name is fine.

Reply:
> But maybe just using the implementing class name is fine.

That was my though, too.